dashboard/app/helpers/lesson_import_helper.rb

```diff
@@ -0,0 +1,213 @@
+      @lesson_activity
+    end
+
+    # Create a lesson with all the levels in them
+    # TODO use the [code-studio] syntax from CB instead
+    @lesson_activity = LessonActivity.new
+    @lesson_activity.name = "Levels"
+    @lesson_activity.lesson_id = lesson_id
+    @lesson_activity.seeding_key = SecureRandom.uuid
+    @lesson_activity.position = activities.length + 1
+    @lesson_activity.save!
+    @lesson_activity.reload
+    @lesson_activity.activity_sections = [create_activity_section_with_level_ranges(0, levels.count - 1, levels)]
+    activities.push(@lesson_activity)
+    activities.flatten
+  end
+
+  # https://github.com/code-dot-org/curriculumbuilder/blob/57cad8f62e50b03e4f16bf77cd9e2e1da5c3e44e/curriculumBuilder/codestudio.py
+  def self.find_skippable_syntax(markdown)
+    regex = /\[code-studio[\d \-]*\]|\[\/?guide\]/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  def self.find_slides(markdown)
+    regex = /^slide!!!slide-\d+(?:<!-- place where you'd like the icon -->)?/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  def self.create_activity_section_with_level_ranges(range_start, range_end, levels)
+    activity_section = ActivitySection.new
+    activity_section.seeding_key ||= SecureRandom.uuid
+    activity_section.position = 0
+    activity_section.lesson_activity = @lesson_activity
+    activity_section.lesson_activity_id = @lesson_activity.id
+    activity_section.save!
+    unless levels.empty?
+      sl_data = levels.slice(range_start..range_end)
+
+      sl_data = sl_data.select {|sl| !sl["inActivitySection"]}
+      activity_section.update_script_levels(sl_data) unless sl_data.blank?
+      sl_data.each {|sl| sl["inActivitySection"] = true}
+    end
+    activity_section
+  end
+
+  def self.find_remarks(markdown)
+    regex = /^!!! *say\s+?[\n\r]*([\d\D]+?)(?=(^\S|$))/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  def self.create_activity_section_with_remark(match)
+    activity_section = ActivitySection.new
+    activity_section.remarks = true
+    description = match[1].strip
+    slide_matches = find_slides(description)
+    if slide_matches.empty? || description.index(slide_matches[0][0]) != 0
+      activity_section.description = description.strip
+    else
+      activity_section.description = description.delete_prefix(slide_matches[0][0])
+      activity_section.slide = true
+    end
+    activity_section
+  end
+
+  # https://github.com/code-dot-org/remark-plugins/blob/master/src/tip.js
+  def self.find_tips(markdown)
+    regex = /^!!! *?([\w-]+)(?: "(.*?)")?(?: <(.*?)>)?(?:[\s]+$)+([\d\D]+?)(?=(^\S|^$))/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  # https://github.com/code-dot-org/remark-plugins/blob/master/src/tiplink.js
+  def self.find_tip_links(markdown)
+    regex = /^([\w-]+)!!! ?([\w-]+)(?:<!-- place where you'd like the icon -->)?(.+?)$/
```

Comment:
have you run into any where the comment was removed? I used to remove the comment sometimes

Reply:
If you're thinking of the `<!-- place where you'd like the icon -->` bit, yeah it is removed sometimes. The `?` at the end makes it optional, so this should match whether or not it's there. I have to write it out explicitly like this so that we can get the capture after it is correct.

Reply:
Ahhh . I'm really bad with regular expressions. That makes sense

Reply:
I should comment these better - these were created after looking at a number of sources plus using a regex checker, so they could use a bit of comments. I'll make a pass.